src/atoms/Checkbox/Checkbox.native.js

```diff
@@ -0,0 +1,166 @@
+import React, { useState, useCallback, useContext } from 'react';
+import PropTypes from 'prop-types';
+import { Text, TouchableOpacity, View } from 'react-native';
+import styled, { ThemeContext } from 'styled-components/native';
+import CheckBoxIcon from './CheckboxIcon';
+
+const CheckboxContainer = styled.View`
+  display: flex;
+  flex-direction: row;
+`;
+
+const RippleView = styled(View)(
+  (props) => `
+  width: ${props.width}px;
+  height: ${props.height}px;
+  border-radius: ${props.width / 2}px;
+  background-color: ${props.underlayColor ? props.underlayColor : 'transparent'};
+`,
+);
+
+const TextContainer = styled.View`
+  margin-left: 4px;
+`;
+
+const Title = styled(Text)(
+  (props) => `
+  color: ${props.disabled ? props.theme.colors.shade[500] : props.theme.colors.shade[700]};
+  font-size: ${props.theme.fonts.size[props.size]};
+`,
+);
+
+const mapSizeToRippleViewProps = (checkBoxSize) => {
+  switch (checkBoxSize) {
+    case 'large':
+      return {
+        width: 24,
+        height: 24,
+      };
+    case 'medium':
+      return {
+        width: 20,
+        height: 20,
+      };
+    case 'small':
+      return {
+        width: 16,
+        height: 16,
+      };
+    case 'xsmall':
+      return { width: 12, height: 12 };
+    default:
+      return {
+        width: 24,
+        height: 24,
+      };
+  }
+};
+
+const mapSizeToTitleTextProps = (checkboxSize) => {
+  switch (checkboxSize) {
+    case 'large':
+      return { size: 'l' };
+    case 'medium':
+      return { size: 'm' };
+    case 'small':
+      return { size: 'xs' };
+    default:
+      return { size: 's' };
+  }
+};
+
+const mapSizetoHelpTextProps = (checkboxSize) => {
+  switch (checkboxSize) {
+    case 'large':
+      return { size: 'm' };
+    case 'medium':
+      return { size: 'xs' };
+    default:
+      return { size: 's' };
+  }
+};
+
+const HelpText = styled(Text)(
+  (props) => `
+  color: ${props.disabled ? props.theme.colors.shade[300] : props.theme.colors.shade[500]};
+  font-size: ${props.theme.fonts.size[props.size]};
+`,
+);
+
+const Checkbox = ({ onClick, checked, disabled, size, title, helpText, testID }) => {
```

Comment:
```suggestion
const Checkbox = ({ onClick, defaultChecked, disabled, size, title, helpText, testID }) => {
```

We're not going to do props reflow for change in the `checked` prop so better call it `defaultChecked`.